migration/src/main/java/kin/sdk/migration/AccountBurner.java

```diff
@@ -0,0 +1,135 @@
+package kin.sdk.migration;
+
+import static kin.sdk.migration.Utils.MAX_RETRIES;
+
+import android.support.annotation.NonNull;
+import kin.sdk.Logger;
+import kin.sdk.migration.bi.IMigrationEventsListener.BurnReason;
+import kin.sdk.migration.bi.IMigrationEventsListener.CheckBurnReason;
+import kin.sdk.migration.core_related.KinAccountCoreImpl;
+import kin.sdk.migration.exception.AccountNotActivatedException;
+import kin.sdk.migration.exception.AccountNotFoundException;
+import kin.sdk.migration.exception.MigrationFailedException;
+import kin.sdk.migration.exception.OperationFailedException;
+import kin.sdk.migration.interfaces.ITransactionId;
+import org.stellar.sdk.responses.HttpResponseException;
+
+class AccountBurner {
+
+	private final MigrationEventsNotifier eventsNotifier;
+
+	AccountBurner(MigrationEventsNotifier eventsNotifier) {
+		this.eventsNotifier = eventsNotifier;
+	}
+
+	/**
+	 * Start the burn account process which checks if the account was already burned and if not then burned it.
+	 *
+	 * @param account is the kin account.
+	 * @return the reason.
+	 */
+	@NonNull
+	BurnReason startBurnAccountProcess(final KinAccountCoreImpl account)
```

Comment:
can be called simply `start` or `burn`  the class name already hint what it does